---
layout: default
---
<div class="home">
  <div class="materials-wrap">
    <p>Additional course websites:</p>
    <ul>
        <li><a href="https://stellar.mit.edu/S/course/6/sp19/6.874/">MIT Stellar</a></li>
        <li><a href="https://stellar.mit.edu/S/course/6/sp19/6.874/gradebook3/">Learning Modules</a> (where you download problem set assignments and upload your solutions)</li>
        <li><a href="https://piazza.com/mit/spring2019/6802j6874j20390j20490jhst506j">Piazza</a> (discussion forum)</li>
    </ul>

    <h2 class="module-header">Course description</h2>
    <p>Welcome to a new approach to computational problems in the life sciences. This subject is not an encyclopedic summary of contemporary methods in systems biology and genomics. Rather, we will explore both conventional and deep learning approaches to key problems in the life sciences, comparing and contrasting their power and limits.</p>

    <p>Our aim is to enable you to evaluate a wide variety of solutions to key problems you will face in this rapidly developing field, and enable you to execute on new enabling solutions that can have large impact.</p>
    
    <p>As part of the subject you will become an expert in using modern cloud resources to implement your solutions to challenging problems, first in problem sets that span a carefully chosen set of tasks, and then in an independent project. You will be programming using Python 3 and TensorFlow 1.12 in Jupyter Notebooks on the Google Cloud, a nod to the importance of carefully documenting your work so it can be precisely reproduced by others.</p>

	<h2 class="module-header">Syllabus and schedule</h2>
    <table class="table">
        <tr class="active">
            <th>&nbsp;</th><th>When&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</th><th>Where&nbsp;&nbsp;&nbsp;&nbsp;</th><th>Description</th><th>Course materials</th><th>Reference</th>
        </tr>
        <tr>
            <td>Lecture 1</td>
            <td>Feb 05 1pm</td>
            <td>32-155</td>
            <td>Machine learning in the computational life sciences</td>
            <td>
                <ul>
                    <li>Read <a href="http://www.deeplearningbook.org/">Goodfellow</a> Chapter 1</li>
                    <li><a href="assets/slides/6.874-lecture-1-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="https://arxiv.org/pdf/1603.06430.pdf">DL in Bioinformatics</a></li>
                    <li><a href="http://msb.embopress.org/content/msb/12/7/878.full.pdf">DL for computational biology</a></li>
                    <li><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1000809">The Roots of Bioinformatics</a></li>
                    <li><a href="http://ieeexplore.ieee.org/stamp/stamp.jsp?reload=true&arnumber=7347331">ML in Genomic Medicine</a></li>
                    <li><a href="https://github.com/gokceneraslan/awesome-deepbio">Awesome DeepBio</a></li>
                    <li><a href="http://colah.github.io/posts/2015-09-Visual-Information/">Visual Information Theory</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Lecture 2</td>
            <td>Feb 07 1pm</td>
            <td>32-155</td>
            <td>Optimizing Feedforward Networks</td>
            <td>
                <ul>
                    <li>Read <a href="http://www.deeplearningbook.org/">Goodfellow</a> Chapter 6</li>
                    <li><a href="assets/slides/FeedForwardBackprop.pdf">Feed Forward Backprop</a></li>
                    <li><a href="assets/slides/6.874-lecture-2-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="https://github.com/aymericdamien/TensorFlow-Examples">Damien</a></li>
                    <li><a href="http://nicklocascio.com/tensorflow-crash-course">Nick Locascio</a></li>
                    <li><a href="https://www.tensorflow.org/versions/r0.11/tutorials/index.html">TF site tutorials</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Recitation 1</td>
            <td>Feb 07 4pm<br/>Feb 08 4pm</td>
            <td>36-156</td>
            <td>Machine learning overview</td>
            <td>
		        <ul>
                    <li><a href="assets/recitations/recitation1-2019.pdf">Recitation slides</a></li>
                </ul>
	        </td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 3</td>
            <td>Feb 12 1pm</td>
            <td>32-155</td>
            <td>Convolutional and recurrent neural networks</td>
            <td>
                <ul>
                    <li>Read <a href="http://www.deeplearningbook.org/">Goodfellow</a> Chapter 9 and 10</li>
                    <li><a href="assets/slides/6.874-lecture-3-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 4</td>
            <td>Feb 14 1pm</td>
            <td>32-155</td>
            <td>Introduction to protein-DNA interactions and ChIP-seq motif discovery</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-4-2019.pdf">Lecture slides</a></li>
                    <li><a href="https://www.youtube.com/watch?v=9_ggu8PKuLs">Manolis Kellis's lecture on Regulatory Genomics</a></li>
                    <li><a href="https://www.youtube.com/watch?v=qaalIf6bn1k">Manolis Kellis's lecture on Epigenomics and ChIP-seq</a></li>
                </ul>
            </td>
            <td>
		        <ul>
                    <li><a href="https://www.nature.com/articles/nrg2641">Peter Park ChIP-seq paper</a></li>
                    <li><a href="https://www.nature.com/articles/nrg3306">Terrence Furey ChIP-seq paper</a></li>
                </ul>
	        </td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Feb 14 11:59pm</td>
            <td></td>
            <td>PS1 due</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 2</td>
            <td>Feb 14 4pm<br />Feb 15 4pm</td>
            <td>36-156</td>
            <td>Neural networks</td>
            <td>
		        <ul>
                    <li><a href="assets/recitations/recitation2-2019.pdf">Recitation slides</a></li>
                    <li><a href="https://www.coursera.org/lecture/convolutional-neural-networks/convolutions-over-volume-ctQZz">Andrew Ng's CNN lectures</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr class="info">
            <td>No class</td>
            <td>Feb 19</td>
            <td></td>
            <td>President’s Day</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 5</td>
            <td>Feb 21 1pm</td>
            <td>32-155</td>
            <td>ML model interpretation  (gradient methods, SIS)</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-5-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td>
                <ul style="font-size: smaller">
                    <li><a href="assets/misc/binder.pdf">Binder et al. (Relevance Propagation)</a></li>
                    <li><a href="assets/misc/dumoulin.pdf">Dumoulin and Visin (Convolution Arithmetic)</a></li>
                    <li><a href="assets/misc/finnegan.pdf">Finnegan and Song (Maximum entropy methods)</a></li>
                    <li><a href="assets/misc/lundberg.pdf">Lundberg and Lee (SHAP)</a></li>
                    <li><a href="assets/misc/ribeiro.pdf">Ribeiro (LIME)</a></li>
                    <li><a href="assets/misc/selvaraju.pdf">Selvaraju et al. (Grad-CAM)</a></li>
                    <li><a href="assets/misc/shrikumar.pdf">Shrikumar et al. (Learning Important Features)</a></li>
                    <li><a href="assets/misc/shrikumar-2.pdf">Shrikumar et al. (DeepLIFT)</a></li>
                    <li><a href="assets/misc/simonyan.pdf">Simonyan et al. (Saliency Maps)</a></li>
                    <li><a href="assets/misc/springenberg.pdf">Springenberg et al. (CNN)</a></li>
                    <li><a href="assets/misc/sundararajan.pdf">Sundararajan et al. (Axiomatic Attribution)</a></li>
                    <li><a href="assets/misc/yosinski.pdf">Yosinski et al. (Deep Visualization)</a></li>
                    <li><a href="assets/misc/zeiler.pdf">Zeiler et al. (Deconvolutional Networks)</a></li>
                    <li><a href="assets/misc/zeiler-2.pdf">Zeiler and Fergus (Understanding Convolutional Networks)</a></li>
                    <li><a href="assets/misc/zhou.pdf">Zhou et al. (Discriminative Localization)</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Recitation 3</td>
            <td>Feb 21 4pm<br />Feb 22 4pm</td>
            <td>36-156</td>
            <td>Motifs, ChIP-seq, models</td>
            <td>
		        <ul>
                    <li><a href="assets/recitations/Recitation3-2019.pdf">Recitation slides</a></li>
                </ul>
		    </td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 6</td>
            <td>Feb 26 1pm</td>
            <td>32-155</td>
            <td>PCA, dimensionality reduction (t-SNE), autoencoders</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-6-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="assets/misc/maaten.pdf">van der Maaten and Hinton (t-SNE)</a></li>
                </ul>
            </td>
        </tr>
        <tr> 
            <td>Lecture 7</td>
            <td>Feb 28 1pm</td>
            <td>32-155</td>
            <td>The expressed genome and RNA splicing (RNA-seq)</td>
            <td>
		        <ul>
                    <li><a href="assets/slides/6.874-lecture-7-2019.pdf">Lecture slides</a></li>
                </ul>
	        </td>
            <td>
                <ul>
                    <li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8">RNA-seq data analysis</a></li>
                    <li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0702-5">Comparative assessment of methods</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Recitation 4</td>
            <td>Feb 28 4pm<br /><span style="color: red">Feb 28 5pm</span></td>
            <td>36-156</td>
            <td>Model interpretation</td>
            <td>
                <ul>
                    <li><a href="assets/recitations/recitation4-2019.pptx">Recitation slides</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 8</td>
            <td>Mar 05 1pm</td>
            <td>32-155</td>
            <td>scRNA seq and cell labeling</td>
            <td>
		        <ul>
                    <li><a href="assets/slides/6.874-lecture-8-2019.pdf">Lecture slides</a></li>
		            <li><a href="assets/BIC_Scores.pdf">BIC scores</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="assets/monocle-cole.pdf">Monocle-cole</a></li>
                    <li><a href="assets/scRNA-disease.pdf">scRNA-disease</a></li>
                    <li><a href="assets/scRNA-integration.pdf">scRNA-integration</a></li>
                    <li><a href="assets/scRNA-pipelines.pdf">scRNA-pipelines</a></li>
                    <li><a href="assets/scRNA-umi.pdf">scRNA-umi</a></li>
                </ul>
	        </td>
        </tr>
        <tr>
            <td>Lecture 9</td>
            <td>Mar 07 1pm</td>
            <td>32-155</td>
            <td>Chromatin accessibility</td>
            <td>
		        <ul>
                    <li><a href="assets/slides/6.874-lecture-9-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="assets/misc/l9/hashimoto.pdf">Hashimoto et al.</a></li>
                    <li><a href="assets/misc/l9/sherwood.pdf">Sherwood et al.</a></li>
                    <li><a href="assets/misc/l9/kelley.pdf">Kelley et al.</a></li>
                </ul>
            </td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Mar 07 11:59pm</td>
            <td></td>
            <td>PS2 due</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 5</td>
            <td>Mar 07 4pm<br />Mar 08 4pm</td>
            <td>36-156</td>
            <td>Dimensionality reduction</td>
            <td>
                <ul>
                    <li><a href="assets/recitations/recitation5-2019.pdf">Recitation slides</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 10</td>
            <td>Mar 12 1pm</td>
            <td>32-155</td>
            <td>Chromatin marks and 3D genome interactions</td>
            <td>
		        <ul>
                    <li><a href="assets/slides/6.874-lecture-10-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="assets/misc/l10/chia-pet-paper.pdf">ChIA-PET</a></li>
                    <li><a href="assets/misc/l10/hichip-paper.pdf">HiChIP</a></li>
                    <li><a href="assets/misc/l10/cid-paper.pdf">CID</a></li>
                    <li><a href="assets/misc/l10/deep-sea.pdf">DeepSEA</a></li>
                    <li><a href="assets/misc/l10/target-finder-paper.pdf">TargetFinder</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Lecture 11</td>
            <td>Mar 14 1pm</td>
            <td>32-155</td>
            <td>Model Uncertainty and Experiment Design (Bayesian Optimization)</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-11-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 6</td>
            <td>Mar 14 4pm<br />Mar 15 4pm</td>
            <td>36-156</td>
            <td>Model uncertainty</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 12</td>
            <td>Mar 19 1pm</td>
            <td>32-155</td>
            <td>Project meetings in class</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-12-2019.pdf">Lecture slides</a></li>
                    <li><a href="assets/final-project-2019.pdf">Project description</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 13</td>
            <td>Mar 21 1pm</td>
            <td>32-155</td>
            <td>Generative Models: gradients, VAEs, GANs</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-13-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="assets/misc/l13/autoencoding-varational-bayes.pdf">Auto-Encoding Variational Bayes</a></li>
                    <li><a href="assets/misc/l13/cyclegans.pdf">Cycle-Consistent Adversarial Networks</a></li>
                    <li><a href="assets/misc/l13/gans.pdf">Generative Adversarial Nets</a></li>
                    <li><a href="assets/misc/l13/Seq2betterSeq.pdf">Sequence to Better Sequence</a></li>
                    <li><a href="assets/misc/l13/wgan.pdf">Wasserstein GAN</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Recitation 7</td>
            <td>Mar 21 4pm<br />Mar 22 4pm</td>
            <td>36-156</td>
            <td>Hyperparameter optimization</td>
            <td>
                <ul>
                    <li><a href="assets/recitations/recitation7-2019.pdf">Recitation slides</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr class="info">
            <td>No class</td>
            <td>Mar 26</td>
            <td></td>
            <td>Spring break</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="info">
            <td>No class</td>
            <td>Mar 30</td>
            <td></td>
            <td>Spring break</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 14</td>
            <td>Apr 02 1pm</td>
            <td>32-155</td>
            <td>Generative models and Therapeutic design</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-14-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td>
                <ul>
                    <li><a href="assets/misc/l14/elmo-model.pdf">Deep contextualized word representations</a></li>
                    <li><a href="assets/misc/l14/gated-graph-nets.pdf">Gated Graph Sequence Neural Networks</a></li>
                    <li><a href="assets/misc/l14/molecule-net.pdf">MoleculeNet</a></li>
                    <li><a href="assets/misc/l14/potentialnet-acs.pdf">PotentialNet</a></li>
                </ul>
            </td>
        </tr>
        <tr>
            <td>Lecture 15<br />by Adrian Dalca</td>
            <td>Apr 04 1pm</td>
            <td>32-155</td>
            <td>Imaging and genotype to phenotype</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-15-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Apr 04 11:59pm</td>
            <td></td>
            <td>PS3 due</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>Recitation 8</td>
            <td>Apr 04 4pm<br />Apr 05 4pm</td>
            <td>36-156</td>
            <td>Protein structure prediction</td>
            <td>
                <ul>
                    <li><a href="assets/recitations/recitation8-2019.pptx">Recitation slides</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr>
            <td>Lecture 16<br />by Haoyang Zeng</td>
            <td>Apr 09 1pm</td>
            <td>32-155</td>
            <td>eQTL prediction and variant prioritization</td>
            <td>
                <ul>
                    <li><a href="assets/slides/6.874-lecture-16-2019.pdf">Lecture slides</a></li>
                </ul>
            </td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>Apr 11</td>
            <td></td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>Apr 11 11:59pm</td>
            <td></td>
            <td>Project proposal due</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="info">
            <td>No class</td>
            <td>Apr 16</td>
            <td></td>
            <td>Patriots' Day</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>Apr 18 1pm</td>
            <td>32-155</td>
            <td>Exam review with TAs</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="danger">
            <td>Exam</td>
            <td>Apr 23 1pm</td>
            <td>32-155</td>
            <td>Exam</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>Apr 25</td>
            <td></td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>Apr 30</td>
            <td></td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 2</td>
            <td></td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 7</td>
            <td></td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 9</td>
            <td></td>
            <td>Project work</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>May 13 11:59pm</td>
            <td></td>
            <td>Project report due</td>
            <td></td>
            <td></td>
        </tr>
        <tr class="warning">
            <td>Deadline</td>
            <td>May 13 11:59pm</td>
            <td></td>
            <td>Project presentation due</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 14 1pm</td>
            <td>32-155</td>
            <td>Project presentations I</td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td></td>
            <td>May 16 1pm</td>
            <td>32-155</td>
            <td>Project presentations II</td>
            <td></td>
            <td></td>
        </tr>
    </table>

    <h2 class="module-header">Tutorials for TensorFlow, NumPy, Google Cloud, and Jupyter notebooks</h2>
    <p>We collected a series of pointers to tutorials on NumPy, TensorFlow, Google Cloud and Conda <a href="tutorials/main/">here</a>. We also provide a <a href="tutorials/quickstart/">Quickstart tutorial</a> to set up essential environment and tools for you to work on problem set 1.</p>
  
    <h2 class="module-header">Prerequisites</h2>
	<p>You should be comfortable with calculus, linear algebra, (Python) programming, probability, and introductory molecular biology. This will be a fast paced course, and it is targeted towards students that are both mathematically and computational capable. There are many other subjects at MIT that teach overviews of computational biology that are less demanding, we would be happy to recommend other options if you find this subject is not what you desire.</p>
    
    <h2 class="module-header">Class meeting times and places</h2>
    <ul>
        <li>Lecture: TR1-2.30 (32-155)</li>
        <li>Recitation: RF4-5 (36-156)</li>
    </ul>

    <h2 class="module-header">Contact</h2>
    <p>You should feel free to contact the lecturer and the TAs about any questions through <a href="mailto:6.874staff@mit.edu">6.874staff@mit.edu</a>. The best way to get detailed questions answered is to attend TA office hours and recitation or post them on <a target="_blank" href="https://piazza.com/mit/spring2019/6802j6874j20390j20490jhst506j">Piazza</a>.</p>

    <h2 class="module-header">Office hours</h2>
    <div class="materials-item">
        David Gifford (gifford@mit.edu): Office hours by appointment
    </div>
    <div class="materials-item">
        Siddhartha Jain (sjX@mit.edu) where X="1": 4pm on Tuesdays (Stata center G5 lounge - right outside of Gates tower elevators on 5th floor)
    </div>
    <div class="materials-item">
        Konstantin Krismer (krismer@mit.edu): 4:30pm on Tuesdays (Stata center G5 lounge)
    </div>
    <div class="materials-item">
        Ge (Saber) Liu (geliu@mit.edu): 5pm on Tuesdays (Stata center G5 lounge)
    </div>

    <h2 class="module-header">Grading</h2>
    <p>Grading will be based upon five programming-intensive problem sets (40%), a comprehensive exam (30%), and a final project (30%). Attendance in lecture is important as the class moves quickly and you will need to be present. There will be an option for some students to give presentations on relevant research papers from the syllabus list. If you opt to present a paper in class it will be graded and substitute for a problem set of your choice. For students enrolled in one of the graduate versions of this class (6.874, 20.490, and HST.506) there will be an extra section on some problem sets. You can use three late days for problem set deadlines (or email from Student Support Services - S3).</p>

    <h2 class="module-header">Project</h2>
    <p>This subject has a substantial deep learning project component. Projects will be completed either as a team of two or individually. 6.874 students must complete an individual project unless permission from the teaching staff is given to work as a team. You are free to choose a problem in the life sciences and develop a deep learning solution using the subject’s cloud resources. Ideas will be provided along with example data, and you can consult the teaching staff to get advice for other ideas or to refine your own ideas. A detailed project description can be found <a target="_blank" href="assets/final-project-2019.pdf">here</a>.</p>

    <h2 class="module-header">Textbook</h2>
    <p>We will be using the book “Deep Learning” by Goodfellow, Bengio, and Courville. You can find the book online <a target="_blank" href="http://deeplearningbook.org">here</a> and <a target="_blank" href="https://github.com/janishar/mit-deep-learning-book-pdf">here</a>. You can purchase a hard copy at <a target="_blank" href="https://mitpress.mit.edu/books/deep-learning">MIT Press</a> or on Amazon.</p>
    <p>Another useful book is the <a href="assets/matrix-cookbook.pdf">Matrix Cookbook</a>, an extensive collection of facts about matrices.</p>
  </div>
</div>
